tests/unittests/OperatorTest.cpp

```diff
@@ -867,6 +867,69 @@ TEST_P(Operator, Gather) {
   EXPECT_FLOAT_EQ(H.at({1, 3, 1}), 1.2);
 }
 
+/// Check if the code generation of transposes
+/// is correct for tensors with 2 dimensions.
+/// Note: This assumes that Tensor::transpose is correct.
+TEST_P(Operator, Transpose2Dims) {
+  auto *A = mod_.createVariable(ElemKind::FloatTy, {20, 13}, "A",
+                                VisibilityKind::Public);
+  A->getPayload().init(Tensor::InitKind::Xavier, 1.0, mod_.getPRNG());
```

Comment:
nit: maybe just randomize here (normally Xavier is the initialization for training path).